src/components/switch/switch.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
```

Comment:
Purely FYI - in our own codebase, we have been trying more and more to move to the Angular Material components (mostly) for these kinds of presentational components. This helps keep styling consistent and reduces our code maintenance costs. So for this, we would use:
https://material.angular.io/components/slide-toggle/overview

Similarly, there are Angular Material cards, tabs, and expansion panels that could be used to replicate other aspects of the dashboard functionality built here.

With that said, writing your own components seems like more fun for this project so the SwitchComponent seems great! Just wanted to share our own practices and let you know about Angular Material in case you didn't know about that option / wanted to try it out.

Reply:
I tried to DIY for learning purposes to understand the data flow of Angular better (since it is my first Angular project that I wrote from scratch), but I definitely agree that using Angular Material would make the project focus more on logic.